### DISCUSSION TITLE
Reflect breaking changes in Azure Data Factory operator, sensor and triggerer

### DISCUSSION THREAD

**Discussion by cesar-vermeulen** [apache/airflow#63196] [Ideas]
### What do you see as an issue?

Since PR https://github.com/apache/airflow/pull/34040/files, factory_name and resource_group_name have become required keyword arguments. This is however not reflected in the documentation:
https://airflow.apache.org/docs/apache-airflow-providers-microsoft-azure/stable/operators/adf_run_pipeline.html

As well as in the docstring (even providing faulty information; we can not use the extra properties in the data factory connection anymore thus leading to confusion):
https://github.com/apache/airflow/blob/522576b9c0c04039d30a01757f0990dbaf72a425/airflow/providers/microsoft/azure/operators/data_factory.py#L91C11-L91C12

### Solving the problem

Adapt documentation so that it becomes clear that resource group name and factory name are now required keyword arguments.

Might need further investigation of why this has become a required keyword arguments whilst keeping the functionality in the hook to default to getting the RG name and factory name from the Airflow connection (https://github.com/apache/airflow/blob/522576b9c0c04039d30a01757f0990dbaf72a425/airflow/providers/microsoft/azure/hooks/data_factory.py#L76)

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
Maybe you would like to do it @cesar-vermeulen ? It's very easy - click "suggest a change on this page" button bottom right on the documentation page you want to modify and you can provide the description in a PR that will be opened for you - without even leaving the GitHub UI. It's even easier to add it than to open the issue you did to be honest (and you do not even need to open issue for that). And great way to contribute back for the free software you use.

**Comment by cesar-vermeulen** [comment]
Hello @potiuk 
I'm more than happy to contribute, however, the reason I created this issue (which I did not explain good enough), was also to determine the reason why https://github.com/apache/airflow/blob/522576b9c0c04039d30a01757f0990dbaf72a425/airflow/providers/microsoft/azure/hooks/data_factory.py#L76 is still being used. To me it makes sense to make these arguments non-required and keep them like they used to be, as this feels more the Airflow-way of things (keeping all connection data in your connection).  
If it is merely adapting the documentation, sure thing :)

**Comment by anupamk36** [comment]
I'm looking to contribute my first open-source PR, would love to have some issues, I see the one above is more or less already closed.

edit : I'm a rookie.

**Comment by potiuk** [comment]
I have no idea as this not my part - but likely asking those who contribute that part ia good idea - I thought it's just documentation issue. I'd say - propose what you think is best @cesar-vermeulen / @anupamk36 and see what people say). Generally we do not expect or even encourage issues for issues like that. PR with proposal is all that it takes - usually such PR is simple and small and discussion on PR with the proposed code is better than on issue, there you can have review comments from those who implemented it or others based on the code seens.

**Comment by eladkal** [comment]
@cesar-vermeulen if you have idea how to address the issue feel free to open a PR
